vendor/k8s.io/kubernetes/pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -295,6 +298,20 @@ func (d *namespacedResourcesDeleter) updateNamespaceStatusFunc(namespace *v1.Nam
 	return d.nsClient.UpdateStatus(&newNamespace)
 }
 
+// updateSetDeletionFailureFunc will set an annotation indicating that the namespace controller has failed to delete all content
+func (d *namespacedResourcesDeleter) updateSetDeletionFailureFunc(err error) func(namespace *v1.Namespace) (*v1.Namespace, error) {
+	errorAnnotationName := "namespace-controller.kube-controller-manager.kubernetes.io/deletion-error"
```

Comment:
before merge, I need to update this to be an alpha something in openshift.io for when the real status pull gets merged upstream